components/BottomNavigation/examples/MDCBottomNavigationBarController.m

```diff
@@ -0,0 +1,265 @@
+
+@end
+
+@implementation MDCBottomNavigationBarController
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _navigationBar = [[MDCBottomNavigationBar alloc] init];
+    _content = [[UIView alloc] init];
+    _viewControllers = @[];
+    _selectedIndex = NSNotFound;
+  }
+
+  return self;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  _navigationBar.delegate = self;
+
+  // Add subviews and create their constraints
+  [self.view addSubview:_navigationBar];
+  [self.view addSubview:_content];
+  [self loadConstraints];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self updateNavigationBarHeight];
+}
+
+- (void)viewSafeAreaInsetsDidChange {
+  [super viewSafeAreaInsetsDidChange];
+  [self updateNavigationBarHeight];
+}
+
+- (void)setSelectedViewController:(nullable UIViewController *)selectedViewController {
+  // Assert that the given VC is one of our view controllers or it is nil (we are unselecting)
+  NSAssert(
+      selectedViewController == nil || [_viewControllers containsObject:selectedViewController],
+      @"Attempting to set BottomBarViewControllers to a view controller it does not contain");
+
+  // Early return if we are already set to the given VC
+  if (_selectedViewController == selectedViewController) {
+    return;
+  }
+
+  // Remove current VC and add new one.
+  [self removeContentViewController:_selectedViewController];
+  [self addContentViewController:selectedViewController];
+
+  // Set the iVar and update selected index
+  _selectedViewController = selectedViewController;
+  self.selectedIndex = [_viewControllers indexOfObject:selectedViewController];
+}
+
+- (void)setSelectedIndex:(NSUInteger)selectedIndex {
+  // If we are setting to NSNotFound deselect the items
+  if (selectedIndex == NSNotFound) {
+    [self deselectCurrentItem];
+    return;
+  }
+
+  BOOL outOfBounds =
+      selectedIndex >= [_viewControllers count] || selectedIndex >= [_navigationBar.items count];
+
+  NSAssert(!outOfBounds,
+           @"Attempting to set BottomBarViewController's selectedIndex to %li. This"
+            " value is not within the bounds of the navigation bar's items and/or view controllers",
+           selectedIndex);
+
+  // Early return if we are out of bounds or if the the index is already selected.
+  if (outOfBounds || selectedIndex == _selectedIndex) {
+    return;
+  }
+
+  // Update the selected index value and views.
+  _selectedIndex = selectedIndex;
+  [self updateViewsForSelectedIndex:selectedIndex];
+}
+
+- (void)setViewControllers:(NSArray<UIViewController *> *)viewControllers {
+  [self deselectCurrentItem];
+  NSMutableArray<UITabBarItem *> *tabBarItems = [NSMutableArray array];
+  for (UIViewController *viewController in viewControllers) {
+    [tabBarItems addObject:viewController.tabBarItem];
+  }
+
+  _navigationBar.items = tabBarItems;
+  _viewControllers = viewControllers;
```

Comment:
Please make this an immutable copy so callers cannot modify the array underneath the VC.